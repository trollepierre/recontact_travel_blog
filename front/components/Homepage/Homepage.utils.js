import { IS_DESKTOP, LARGE_DESKTOP_LIMIT } from '~/services/utils/responsive/responsive-utils';
import { screenWidth } from '~/services/utils/screen/screen-utils';

const LARGE_WIDTH_ARTICLE_CARDS_NUMBER = 4
const SMALL_WIDTH_ARTICLE_CARDS_NUMBER = 3
const MAP_SIZE = 960
const ARTICLE_CARD_SIZE = 300
const NUMBER_OF_ARTICLE_CARDS_ROW = 2

export const articlesNumberLimit = () => {
  if (!IS_DESKTOP()) {
    return LARGE_WIDTH_ARTICLE_CARDS_NUMBER
  }
  if (screenWidth() <= LARGE_DESKTOP_LIMIT) {
    return SMALL_WIDTH_ARTICLE_CARDS_NUMBER
  }
  return Math.floor((screenWidth() - MAP_SIZE) / ARTICLE_CARD_SIZE) * NUMBER_OF_ARTICLE_CARDS_ROW
}
