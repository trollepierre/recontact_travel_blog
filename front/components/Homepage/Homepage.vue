<template>
  <div class="homepage">
    <app-header/>
    <main class="main">
      <h1 class="h1">
        {{  $t("mapHeadline") }}
      </h1>
      <world-map/>
    </main>
  </div>
</template>

<script>
  import WorldMap from './Map/Map.vue'
  import AppHeader from '../AppHeader/AppHeader.vue'

  export default {
    components: {
      AppHeader,
      WorldMap,
    },
    i18n: {
      silentTranslationWarn: true,
      messages: {
        fr: {
          mapHeadline: 'Retrouvez nos articles à travers le monde :',
        },
        en: {
          mapHeadline: 'Retrieve our articles all around the world:',
        },
      },
    },
  }
</script>

<style scoped>
.h1 {
  margin-bottom: 30px;
  font-size: 20px;
}

.homepage {
  display: flex;
  justify-content: center;
}

.main {
  margin-top: 60px;
}

@media only screen and (min-width: 650px) {
  .h1 {
    font-size: 32px;
  }
}
</style>
