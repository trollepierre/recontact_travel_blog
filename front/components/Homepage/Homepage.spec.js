import VueI18n from 'vue-i18n'
import Homepage from './Homepage.vue'

jest.mock('@/components/Homepage/Map/Map', () => () => '<div>Map</div>')

describe('Component | Homepage.vue', () => {
  let localVue
  let wrapper

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(VueI18n)
    wrapper = shallowMount(Homepage, { localVue })
  })

  describe('template', () => {
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
