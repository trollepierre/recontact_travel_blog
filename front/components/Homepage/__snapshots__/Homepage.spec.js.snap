// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | Homepage.vue template should match snapshot 1`] = `
<div class="homepage">
  <app-header-stub></app-header-stub>
  <main class="main">
    <h1 class="h1">
      Retrieve our articles all around the world:
    </h1>
    <!---->
  </main>
</div>
`;
